cassandra/io/eventletreactor.py

```diff
@@ -53,19 +52,39 @@ class EventletConnection(Connection):
     _write_watcher = None
     _socket = None
 
+    _timers = None
+    _timeout_watcher = None
+    _new_timer = None
+
     @classmethod
     def initialize_reactor(cls):
         eventlet.monkey_patch()
+        if not cls._timers:
+            cls._timers = TimerManager()
+            cls._timeout_watcher = eventlet.spawn(cls.service_timeouts)
+            cls._new_timer = Event()
+
+    @classmethod
+    def create_timer(cls, timeout, callback):
+        timer = Timer(timeout, callback)
+        cls._timers.add_timer(timer)
+        cls._new_timer.set()
+        return timer
+
+    @classmethod
+    def service_timeouts(cls):
+        timer_manager = cls._timers
+        while True:
+            next_end = timer_manager.service_timeouts()
+            sleep_time = max(next_end - time.time(), 0) if next_end else 10000
+            cls._new_timer.wait(sleep_time)
```

Comment:
A comment to explain roughly how this loop works would be good -- took me a minute to figure out.